sonaranalyzer-dotnet/its/regression-test.ps1

```diff
@@ -53,6 +53,39 @@ function Build-Project([string]$ProjectName, [string]$SolutionRelativePath, [int
         /flp:"logFile=output\${ProjectName}.log;verbosity=d"
 }
 
+function Build-Project-DotnetTool([string]$ProjectName, [string]$SolutionRelativePath) {
+    if ($project -And -Not ($ProjectName -eq $project)) {
+        Write-Host "Build skipped: $ProjectName"
+        return
+    }
+
+    New-Item -ItemType directory -Path .\output\$ProjectName | out-null
+
+    $solutionPath = Resolve-Path ".\sources\${ProjectName}\${SolutionRelativePath}"
+
+    # The PROJECT env variable is used by 'SonarAnalyzer.Testing.ImportBefore.targets'
+    Write-Debug "Setting PROJECT environment variable to '${ProjectName}'"
+    $Env:PROJECT = $ProjectName
+
+    # TODO create the global.json based on target SDK version (passed by parameter) and then clean it in this function
+    $globalJsonPath = Resolve-Path ".\sources\${ProjectName}\global.json"
+    Copy-Item $globalJsonPath .
+
+    # The PROJECT env variable is used by 'SonarAnalyzer.Testing.ImportBefore.targets'
+    Write-Debug "Setting PROJECT environment variable to '${ProjectName}'"
+
+    $Env:PROJECT = $ProjectName
+    dotnet restore $solutionPath
```

Comment:
When using `dotnet` tool you don't have to call `restore` since this runs implicitly. More details here: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build#implicit-restore

Reply:
(I am using `dotnet msbuild` which doesn't do all the magic of `dotnet build`)